internals/review_models.py

```diff
@@ -93,13 +103,26 @@ def set_approval(cls, feature_id, field_id, new_state, set_by_email):
     new_appr.put()
     logging.info('new_appr is %r', new_appr.key.integer_id())
 
+    # Write for new Vote entity.
+    new_vote = Vote(
+        id=new_appr.key.integer_id(), feature_id=feature_id, gate_id=field_id,
+        state=new_state, set_on=now, set_by=set_by_email)
+    new_vote.put()
+    logging.info('new_appr is %r', new_vote.key.integer_id())
+
   @classmethod
   def clear_request(cls, feature_id, field_id):
     """After the review requirement has been satisfied, remove the request."""
     review_requests = cls.get_approvals(
         feature_id=feature_id, field_id=field_id, states=[cls.REVIEW_REQUESTED])
     for rr in review_requests:
       rr.key.delete()
+    
+    # Delete associated Vote entities as well.
```

Comment:
This will be phased out later, but it is fine for now.  Eventually, we will keep all Vote entities and set the state of the Gate instead of deleting the review request.